### ISSUE TITLE
[Bug][Insets] navigationBarsWithImePadding() (imePadding() also) doesn't work with BoxWithConstraints

### ISSUE THREAD

**Issue by yanzm** [google/accompanist#421]
## Describe the bug
navigationBarsWithImePadding() (imePadding() also) doesn't work with BoxWithConstraints() and specifying windowInsetsAnimationsEnabled for ProvideWindowInsets().

## To Reproduce

```kotlin
/**
 * BoxWithConstraints() and ProvideWindowInsets(windowInsetsAnimationsEnabled = false)
 */
class BoxWithConstraintsActivity2 : ComponentActivity() {

    @OptIn(ExperimentalAnimatedInsets::class)
    override fun onCreate(savedInstanceState: Bundle?) {
        WindowCompat.setDecorFitsSystemWindows(window, false)
        super.onCreate(savedInstanceState)

        setContent {
            MaterialTheme {
                ProvideWindowInsets(windowInsetsAnimationsEnabled = false) {

                    BoxWithConstraints(
                        modifier = Modifier
                            .fillMaxSize()
                            .statusBarsPadding()
                            .navigationBarsWithImePadding()
                            .padding(24.dp)
                    ) {
                        var text by remember { mutableStateOf("BoxWithConstraints() and ProvideWindowInsets(windowInsetsAnimationsEnabled = false)") }

                        TextField(
                            value = text,
                            onValueChange = { text = it },
                            modifier = Modifier
                                .fillMaxSize()
                        )
                    }
                }
            }
        }
    }
}
```

| | Box | BoxWithConstraints |
|-|-|-|
| not specify windowInsetsAnimationsEnabled | work | almost work |
| specify windowInsetsAnimationsEnabled | work | not work |

Sample code : https://github.com/yanzm/NavigationBarsWithImePaddingSample

## Expected behavior
work like Box()

## Screenshots

|Box() and ProvideWindowInsets()|Box() and ProvideWindowInsets(windowInsetsAnimationsEnabled = false)|
|-|-|
|<img src="https://user-images.githubusercontent.com/455504/118588155-d3d98900-b7d8-11eb-9e77-d2cf5790e9d7.png" width=240>|<img src="https://user-images.githubusercontent.com/455504/118588151-d20fc580-b7d8-11eb-8c71-b07ab838e166.png" width=240>|

|BoxWithConstraints() and ProvideWindowInsets()|BoxWithConstraints() and ProvideWindowInsets(windowInsetsAnimationsEnabled = false)|
|-|-|
|<img src="https://user-images.githubusercontent.com/455504/118588158-d4721f80-b7d8-11eb-97ef-a6bf0cc52d82.png" width=240>|<img src="https://user-images.githubusercontent.com/455504/118588160-d50ab600-b7d8-11eb-9683-328da7ddc145.png" width=240>|

## Environment:
 - Android OS version: Android 11
 - Device: Pixel 3
 - Compose version : 1.0.0-beta06
 - Accompanist version: 0.9.1

**Comment by chrisbanes** [comment]
Duplicate of #261.

**Comment by yanzm** [comment]
I confirmed this was fixed with compose_version 1.0.0-beta07 and accompanist version: 0.10.0.
